broker/src/test/java/io/camunda/zeebe/broker/system/partitions/BrokerDifferentRuntimeDirectoryTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.raft.partition.RaftPartition;
+import io.camunda.zeebe.broker.test.EmbeddedBrokerRule;
+import java.nio.file.Paths;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class BrokerDifferentRuntimeDirectoryTest {
```

Comment:
:question: It seems overkill to start the whole broker to test this feature. But I'm not sure how else to test it. Unit testing PartitionFactory is not enough because we don't know if anywhere else runtime location is assumed to be in data directory. If you have other ideas to test please let me know.